ethcore/src/json_tests/chain.rs

```diff
@@ -17,33 +17,119 @@
 use std::path::Path;
 use std::sync::Arc;
 use client::{Client, ClientConfig};
-use client_traits::{ImportBlock, ChainInfo};
+use client_traits::{ImportBlock, ChainInfo, StateOrBlock, Balance, Nonce, BlockChainClient};
 use spec::Genesis;
-use ethjson::test_helpers::blockchain;
+use ethjson::{
+	test_helpers::blockchain,
+	spec::State
+};
 use miner::Miner;
 use io::IoChannel;
 use test_helpers::{self, EvmTestClient};
-use types::verification::Unverified;
+use types::{
+	verification::Unverified,
+	ids::BlockId,
+	client_types::StateResult
+};
 use verification::{VerifierType, queue::kind::BlockLike};
 use super::{HookType, SKIP_TESTS};
+use rustc_hex::ToHex;
+use ethereum_types::{U256, H256}; 
 
 #[allow(dead_code)]
-fn skip_test(name: &String) -> bool {
-	SKIP_TESTS
-		.block
-		.iter()
+fn skip_test(name: &String, is_legacy: bool) -> bool {
+	let skip_set = if is_legacy {
+		&SKIP_TESTS.legacy_block
+	} else {
+		&SKIP_TESTS.block
+	};
+	skip_set.iter()
 		.any(|block_test|block_test.subtests.contains(name))
```

Comment:
```suggestion
		.any(|block_test| block_test.subtests.contains(name))
```

Reply:
I hope that one in the near future we can apply rustfmt to the whole project :sweat_smile: